application/src/main/java/org/opentripplanner/transit/service/TransitService.java

```diff
@@ -64,6 +64,16 @@
  *   copy-on-write and shares a lot of objects with any other TransitLayer instances.
  */
 public interface TransitService {
+  /**
+   * @return empty if the trip doesn't exist in the timetable (e.g. real-time added)
+   */
+  Optional<List<TripTimeOnDate>> getScheduledTripTimes(Trip trip);
+
+  /**
+   * @return empty if the trip doesn't run on the date specified
+   */
+  Optional<List<TripTimeOnDate>> getTripTimeOnDates(Trip trip, LocalDate serviceDate);
```

Comment:
I think it is a valid use case to query an arbitrary pair of trip / service date, given that the trip is valid. As a traveller, I want to see if the trip I am on will still run tomorrow (or if it will be diverted).

We have planned a feature called "save regular journey" in our product, which will be implemented by saving an itinerary and showing the calendar when the legs are scheduled to run.